rct229/rulesets/ashrae9012019/section4/section4rule2.py

```diff
@@ -117,100 +112,96 @@ def get_calc_vals(self, context, data=None):
-                    getattr_(
-                        find_exactly_one(
-                            f'$[*][?(@.id="{maximum_humidity_stpt_sch_id_p}")]',
-                            schedules_p,
-                        ),
-                        "schedules",
-                        "hourly_values",
+                maximum_humidity_stpt_hourly_values_p = None
+                if maximum_humidity_stpt_sch_id_p:
+                    assert_(schedules_p, f"schedules is missing in the RMD: {PROPOSED}")
+                    maximum_humidity_stpt_hourly_values_p = (
+                        getattr_(
+                            find_exactly_one(
+                                f'$[*][?(@.id="{maximum_humidity_stpt_sch_id_p}")]',
+                                schedules_p,
+                            ),
+                            "schedules",
+                            "hourly_values",
+                        )
+                    )
+                    assert_(
+                        len(maximum_humidity_stpt_hourly_values_p) == number_of_hours,
+                        f"maximum humidity setpoint hourly schedule {maximum_humidity_stpt_sch_id_p} have unexpected "
+                        f"number of hours. The hours should be {number_of_hours}, but got "
+                        f"{len(maximum_humidity_stpt_hourly_values_p)} instead. "
                     )
-                    if schedules_p and maximum_humidity_stpt_sch_id_p
-                    else None
-                )
-                assert_(
-                    maximum_humidity_stpt_hourly_values_p is None
-                    or isinstance(maximum_humidity_stpt_hourly_values_p, list)
-                    and len(maximum_humidity_stpt_hourly_values_p) == number_of_hours,
-                    f"maximum humidity setpoint hourly schedule {maximum_humidity_stpt_sch_id_p} have unexpected number of hours. The hours should be {number_of_hours}, but got {len(maximum_humidity_stpt_hourly_values_p)} instead."
-                    if maximum_humidity_stpt_hourly_values_p
-                    else "maximum humidity setpoint hourly schedule does not exist in the proposed model.",
-                )
 
+                # None matches None or hourly values matched exactly.
                 minimum_humidity_schedule_matched = (
                     minimum_humidity_stpt_hourly_values_b
                     == minimum_humidity_stpt_hourly_values_p
                 )
+
+                # ^ comparison for data type.
                 minimum_humidity_schedule_type_matched = (
                     minimum_humidity_stpt_hourly_values_b is None
-                    and minimum_humidity_stpt_hourly_values_p is None
-                ) or (
-                    minimum_humidity_stpt_hourly_values_b is not None
-                    and minimum_humidity_stpt_hourly_values_p is not None
+                    == minimum_humidity_stpt_hourly_values_p is None
```

Comment:
Here we will need a parenthesis. Otherwise, it will return false if both `minimum_humidity_stpt_hourly_values_b ` and `minimum_humidity_stpt_hourly_values_p` exist. That' what happened when I tested locally, but I'm not sure why this does not influence the testing result. 
```
 minimum_humidity_schedule_type_matched = (
                    (minimum_humidity_stpt_hourly_values_b is None)
                    == (minimum_humidity_stpt_hourly_values_p is None)
)
```
Could you double-check? @weilixu Thanks

Reply:
You are right, we need to add parenthesis.